python/src/_dynamic_memory_index.py

```diff
@@ -59,17 +72,79 @@ def from_file(
+        - **max_vectors**: Capacity of the memory index including space for future insertions.
+        - **complexity**: Complexity (a.k.a `L`) references the size of the list we store candidate approximate
+          neighbors in. It's used during save (which is an index rebuild), and it's used as an initial search size to
+          warm up our index and lower the latency for initial real searches.
+        - **graph_degree**: Graph degree (a.k.a. `R`) is the maximum degree allowed for a node in the index's graph
+          structure. This degree will be pruned throughout the course of the index build, but it will never grow beyond
+          this value. Higher R values require longer index build times, but may result in an index showing excellent
+          recall and latency characteristics.
+        - **saturate_graph**: If True, the adjacency list of each node will be saturated with neighbors to have exactly
+          `graph_degree` neighbors. If False, each node will have between 1 and `graph_degree` neighbors.
+        - **max_occlusion_size**: The maximum number of points that can be considered by occlude_list function.
+        - **alpha**: The alpha parameter (>=1) is used to control the nature and number of points that are added to the
+          graph. A higher alpha value (e.g., 1.4) will result in fewer hops (and IOs) to convergence, but probably
+          more distance comparisons.
+        - **num_threads**: Number of threads to use when creating this index. `0` indicates we should use all available
+          logical processors.
+        - **filter_complexity**: Complexity to use when using filters. Default is 0.
+        - **num_frozen_points**: Number of points to freeze. Default is 1.
+        - **initial_search_complexity**: Should be set to the most common `complexity` expected to be used during the
+          life of this `diskannpy.DynamicMemoryIndex` object. The working scratch memory allocated is based off of
+          `initial_search_complexity` * `search_threads`. Note that it may be resized if a `search` or `batch_search`
+          operation requests a space larger than can be accommodated by these values.
+        - **search_threads**: Should be set to the most common `num_threads` expected to be used during the
+          life of this `diskannpy.DynamicMemoryIndex` object. The working scratch memory allocated is based off of
+          `initial_search_complexity` * `search_threads`. Note that it may be resized if a `batch_search`
+          operation requests a space larger than can be accommodated by these values.
+        - **concurrent_consolidation**: If True, the consolidation process will be parallelized. This will result in
+          faster consolidation times, but will also result in higher memory usage during the consolidation process.
+        - **index_prefix**: The prefix of the index files. Defaults to "ann".
```

Comment:
can we possibly default ANN_R{R}_L{L}? OK with either

Reply:
Ugh. This actually makes for even more problems; if we do this when we have R and L, we need to force the user to know what it would be interpolated to on load.  I'd much rather keep it fixed as `ann` and the user can create their own convention, just so we don't make it easy to get out of sync.

The static search classes don't have any idea what R or L is used, since it's immaterial to searching.  Dynamic knows, but it just makes for a more nuanced api that is hard to keep track of